pdal/util/ProgramArgs.hpp

```diff
@@ -356,10 +382,11 @@ class TArg : public Arg
       \param variable  Variable to which the value of the argument should
         be bound.
       \param def  Default value to be assigned to the bound variable.
+      \param required  Is field required or not.
     */
     TArg(const std::string& longname, const std::string& shortname,
-        const std::string& description, T& variable, T def) :
-        Arg(longname, shortname, description), m_var(variable),
+         const std::string& description, T& variable, T def, const bool required) :
+        Arg(longname, shortname, description, required, TypeName<T>::get()), m_var(variable),
```

Comment:
May be, we need to add an overload which allows to add a custom handwritten type here.